internal/util/util.go

```diff
@@ -337,3 +346,75 @@ func contains(s []string, key string) bool {
 
 	return false
 }
+
+// DefaultLog helps in logging with klog.level 1
+func DefaultLog(message string, args ...interface{}) {
+	logMessage := fmt.Sprintf(message, args...)
+	// If logging is disabled, don't evaluate the arguments
+	if klog.V(Default).Enabled() {
+		klog.InfoDepth(1, logMessage)
+	}
+}
+
+// UsefulLog helps in logging with klog.level 2
+func UsefulLog(ctx context.Context, message string, args ...interface{}) {
+	logMessage := fmt.Sprintf(Log(ctx, message), args...)
+	// If logging is disabled, don't evaluate the arguments
+	if klog.V(Useful).Enabled() {
+		klog.InfoDepth(1, logMessage)
+	}
+}
+
+// ExtendedLogMsg helps in logging a message with klog.level 3
+func ExtendedLogMsg(message string, args ...interface{}) {
+	logMessage := fmt.Sprintf(message, args...)
+	// If logging is disabled, don't evaluate the arguments
+	if klog.V(Default).Enabled() {
+		klog.InfoDepth(1, logMessage)
+	}
+}
+
+// ExtendedLog helps in logging with klog.level 3
+func ExtendedLog(ctx context.Context, message string, args ...interface{}) {
+	logMessage := fmt.Sprintf(Log(ctx, message), args...)
+	// If logging is disabled, don't evaluate the arguments
+	if klog.V(Extended).Enabled() {
+		klog.InfoDepth(1, logMessage)
+	}
+}
+
+// DebugLogMsg helps in logging a message with klog.level 4
+func DebugLogMsg(message string, args ...interface{}) {
+	logMessage := fmt.Sprintf(message, args...)
+	// If logging is disabled, don't evaluate the arguments
+	if klog.V(Default).Enabled() {
```

Comment:
can you check this case, what happens when we pass 0,1,2,3,4 and remove this if check? will Trace and Debug will get logged when the loglevel is 0?

Reply:
I haven't added a check for log level 0 as of now as I didn't find it's usage in the repo. I required I will add that too.

Reply:
The If check basically helps us to specify log level along with mentioning depth using InfoDepth. Removing If check will result in losing the track of log level.

Reply:
I went through the klog package, This check is necessary, Thank you